### PR TITLE
[diem framework specs] update specs and remove verify = false from DiemSystem

### PR DESCRIPTION
**Pull request by emmazzz**
## Motivation

This PR removes `verify = false` from DiemSystem, resolves the errors caused by the removal by adding more asserts and abort specs. It also uses the new `choose` operator in some places.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

Yes.

### CODE REVIEW

File: language/diem-framework/modules/DiemSystem.move [hunk 1]
```diff
@@ -290,17 +295,20 @@ module DiemFramework::DiemSystem {
...
-            assert(DiemTimestamp::now_microseconds() >
-                   validator_info.last_config_update_time + FIVE_MINUTES,
-                   ECONFIG_UPDATE_RATE_LIMITED);
+            assert(
+                validator_info.last_config_update_time <= MAX_U64 - FIVE_MINUTES,
+                Errors::limit_exceeded(ECONFIG_UPDATE_TIME_OVERFLOWS)
+            );
+            assert(
+                DiemTimestamp::now_microseconds() > validator_info.last_config_update_time + FIVE_MINUTES,
+                Errors::invalid_state(ECONFIG_UPDATE_RATE_LIMITED)
```
Comment (emmazzz) [inline language/diem-framework/modules/DiemSystem.move:304]: Is invalid state the appropriate category here?

Comment (wrwg) [reply language/diem-framework/modules/DiemSystem.move:304]: LIMIT_EXCEEDED might be better

File: language/diem-framework/modules/DiemSystem.move [hunk 2]
```diff
@@ -318,6 +326,13 @@ module DiemFramework::DiemSystem {
                 v_info.addr == validator_addr
                 && v_info.config != ValidatorConfig::spec_get_config(validator_addr));
         include is_validator_info_updated ==> DiemConfig::ReconfigureAbortsIf;
+        let validator_index =
+            choose i in 0..len(spec_get_validators()) where spec_get_validators()[i].addr == validator_addr;
```
Comment (wrwg) [inline language/diem-framework/modules/DiemSystem.move:330]: There is also a function `range(v)` which delivers the range of a vector.

File: language/diem-framework/modules/ValidatorAdministrationScripts.move [hunk 3]
```diff
@@ -360,7 +361,13 @@ module ValidatorAdministrationScripts {
                         fullnode_network_addresses,
                    });
         include is_validator_info_updated ==> DiemConfig::ReconfigureAbortsIf;
-
+        let validator_index =
+            choose i in 0..len(DiemSystem::spec_get_validators()) where DiemSystem::spec_get_validators()[i].addr == validator_account;
```
Comment (wrwg) [inline language/diem-framework/modules/ValidatorAdministrationScripts.move:365]: 2nd time use of this fragment; perhaps a spec fun?

**Review by wrwg — APPROVED** [review]

**Review by wrwg — DISMISSED** [review]

**Review by wrwg — APPROVED** [review]

**Pull request closed without merging**